pkg/pipeline/pipeline.go

```diff
@@ -600,32 +601,37 @@ func (s AthenaConfig) MarshalJSON() ([]byte, error) {
 	return json.Marshal(s)
 }
 
+type IntervalModifiers struct {
+	Start string `json:"start" yaml:"start" mapstructure:"start"`
+	End   string `json:"end" yaml:"end" mapstructure:"end"`
```

Comment:
I think we should do the parsing logic once and convert these into a proper structure afterwards, meaning that the resulting struct should be something like this:

```go
type TimeModifier struct {
	Days int `json:"days"`
	Hours int `json:"hours"`
	Minutes int `json:"minutes"`
	Seconds int `json:"seconds"`
	CronPeriods int `json:"cron_periods"`
}

func (t *TimeModifier) UnmarshalYAML(value *yaml.Node) error {
	// convert string to time modifier
}

type IntervalModifiers struct {
	Start TimeModifier `json:"start"`
	End   TimeModifier `json:"end"`
}
```

This would allow us to avoid having to repeat this same logic everywhere in our other applications.